Mods/ArchipelagoPlayer.SC2Mod/Base.SC2Data/GameData/EffectData.xml

```diff
@@ -17081,4 +17081,41 @@
     <CEffectRemoveBehavior id="AP_VoidRaySwarmDamageBoostCancel">
         <BehaviorLink value="AP_VoidRaySwarmDamageBoost"/>
     </CEffectRemoveBehavior>
+    <CEffectCreatePersistent id="AP_VoidRayPurifierBeamLaunchPersistent">
+        <EditorCategories value="Race:Protoss"/>
+        <WhichLocation Value="TargetUnit"/>
+        <TimeScaleSource Value="Caster"/>
+        <Flags index="Channeled" value="1"/>
+        <PeriodCount value="1"/>
+        <PeriodicValidator value="CasterNotDead"/>
+        <PeriodicEffectArray value="AP_VoidRayPurifierBeamUseMagazine"/>
+        <PeriodicPeriodArray value="0.1"/>
+    </CEffectCreatePersistent>
+    <CEffectApplyBehavior id="AP_VoidRayPurifierBeamDamageFieldAB">
+        <WhichUnit Value="Target"/>
+        <Behavior value="AP_VoidRayPurifierBeamDamageField"/>
+    </CEffectApplyBehavior>
+    <CEffectEnumArea id="AP_VoidRayPurifierBeamSearch">
+        <AINotifyFlags index="HurtEnemy" value="1"/>
+        <SearchFilters value="Ground;Self,Player,Ally,Neutral,Missile,Stasis,Dead,Hidden,Invulnerable"/>

```

Comment:
Doesn't damage allied primary target.  (if you order to attack your stuff or an ally). Check Viking's shredder rounds that have handled that case
![Snímek obrazovky pořízený 2024-07-13 19-55-42](https://github.com/user-attachments/assets/70f73458-a08b-48f4-84fe-456e154b9aef)

Reply:
Viking piercing rounds do something different from what I need, but I have a workaround I can try; I'll just apply a temporary behavior if the target's friendly, then make the search effect look for targets that either have that behavior or are enemies.

Reply:
Fixed, Dawnbringer beams now damage all enemy units, AND friendly targets that their parent is specifically attacking.